itests/kit/node_unmanaged.go

```diff
@@ -440,66 +379,228 @@ func (tm *TestUnmanagedMiner) waitForMutableDeadline(ctx context.Context, sector
 	}
 }
 
-func (tm *TestUnmanagedMiner) OnboardCCSector(ctx context.Context, proofType abi.RegisteredSealProof) (abi.SectorNumber, chan WindowPostResp,
-	context.CancelFunc) {
-	req := require.New(tm.t)
+func (tm *TestUnmanagedMiner) NextSectorNumber() abi.SectorNumber {
 	sectorNumber := tm.currentSectorNum
 	tm.currentSectorNum++
+	return sectorNumber
+}
+
+func (tm *TestUnmanagedMiner) PrepareSectorForProveCommit(
+	ctx context.Context,
+	proofType abi.RegisteredSealProof,
+	sectorNumber abi.SectorNumber,
+	pieces []abi.PieceInfo,
+) (seedEpoch abi.ChainEpoch, proveCommit []byte) {
+
+	req := require.New(tm.t)
+
+	// Wait for the pre-commitseal randomness to be available (we can only draw seal randomness from tipsets that have already achieved finality)
+	preCommitSealRandEpoch := tm.waitPreCommitSealRandomness(ctx, sectorNumber, proofType)
+
+	// Generate a Pre-Commit for the CC sector -> this persists the proof on the `TestUnmanagedMiner` Miner State
+	tm.generatePreCommit(ctx, sectorNumber, preCommitSealRandEpoch, proofType, pieces)
 
 	// --------------------Create pre-commit for the CC sector -> we'll just pre-commit `sector size` worth of 0s for this CC sector
 
-	// Step 1: Wait for the pre-commitseal randomness to be available (we can only draw seal randomness from tipsets that have already achieved finality)
-	preCommitSealRand := tm.waitPreCommitSealRandomness(ctx, sectorNumber)
+	if !proofType.IsNonInteractive() {
+		// Submit the Pre-Commit to the network
+		var uc *cid.Cid
+		if len(pieces) > 0 {
+			unsealedCid := tm.unsealedCids[sectorNumber]
+			uc = &unsealedCid
+		}
+		r, err := tm.SubmitMessage(ctx, &miner14.PreCommitSectorBatchParams2{
+			Sectors: []miner14.SectorPreCommitInfo{{
+				Expiration:    2880 * 300,
+				SectorNumber:  sectorNumber,
+				SealProof:     proofType,
+				SealedCID:     tm.sealedCids[sectorNumber],
+				SealRandEpoch: preCommitSealRandEpoch,
+				UnsealedCid:   uc,
+			}},
+		}, 1, builtin.MethodsMiner.PreCommitSectorBatch2)
+		req.NoError(err)
+		req.True(r.Receipt.ExitCode.IsSuccess())
+	}
+
+	// Generate a ProveCommit for the CC sector
+	var seedRandomness abi.InteractiveSealRandomness
+	seedEpoch, seedRandomness = tm.proveCommitWaitSeed(ctx, sectorNumber, proofType)
 
+	proveCommit = []byte{0xde, 0xad, 0xbe, 0xef} // mock prove commit
 	if !tm.mockProofs {
-		// Step 2: Write empty bytes that we want to seal i.e. create our CC sector
-		tm.makeAndSaveCCSector(ctx, sectorNumber)
+		proveCommit = tm.generateProveCommit(ctx, sectorNumber, proofType, seedRandomness, pieces)
+	}
+
+	return seedEpoch, proveCommit
+}
+
+func (tm *TestUnmanagedMiner) SubmitProveCommit(
+	ctx context.Context,
```

Comment:
Yeah so we're completely handling `ProveCommits` for NI-PoRep proofs but I am wondering if something is missing with replacing the PC1/PC2 for these proofs.

Reply:
nope, it's that simple; the only change is that we take "circuit proofs" instead - if we didn't take "circuit proofs" then we'd get an aggregate output from C2, but since we changed `ProveCommitSectorsNI` to only accept aggregate we have to collect all circuits from all sectors and aggregate them all together

you can see this work in https://github.com/filecoin-project/lotus/blob/master/cmd/lotus-bench/bench-sectors.sh, one of the experiments I had to work through to get this all figured out